app/components/primer/alpha/segmented-control-element.ts

```diff
@@ -0,0 +1,36 @@
+/* eslint-disable custom-elements/no-constructor */
+export default class SegmentedControlElement extends HTMLElement {
+  constructor() {
+    super()
+
+    this.addEventListener('click', (event: MouseEvent) => {
+      const controls = Array.from(this.querySelectorAll<HTMLElement>('ul button')).filter(
+        tab => tab instanceof HTMLElement && tab.closest(this.tagName) === this
+      ) as HTMLElement[]
+
+      if (!(event.target instanceof Element)) return
+      if (event.target.closest(this.tagName) !== this) return
+
+      const selectedControl = event.target.closest('button')
+      if (!(selectedControl instanceof HTMLElement) || !selectedControl.closest('ul')) return
+
+      for (const control of controls) {
+        control.setAttribute('aria-current', 'false')
+      }
+
+      selectedControl.setAttribute('aria-current', 'true')
+      selectedControl.focus()
+    })
+  }
```

Comment:
@keithamus This seems super generic, Is there an existing behavior or web-component that takes care of this `click -> selected state` action described here? https://primer.style/design/components/segmented-control#keyboard-navigation

Reply:
Might be a good opportunity to bring in Catalyst which would save on boilerplate, and we could work on extracting that kind of behaviour.